infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -445,3 +494,95 @@ private fun verifySwitchSuggestions(
         id to suggestion
     }
 }
+
+private fun findNewLocationTracksForUnusedDuplicates(
+    geocodingContext: GeocodingContext,
+    unusedDuplicates: List<Pair<LocationTrack, LayoutAlignment>>,
+    splitTargetLocationTracks: List<Pair<LocationTrack, LayoutAlignment>>,
+): List<LocationTrack> {
+
+    data class MostOverlappingLocationTrackReference(
+        val locationTrack: LocationTrack? = null,
+        val overlapPercentage: Double = 0.0,
+    )
+
+    val geocodedUnusedDuplicates = unusedDuplicates
+        .mapNotNull { (unusedDuplicate, alignment) ->
+            getAlignmentStartAndEndM(geocodingContext, alignment)?.let { startAndEnd ->
+                unusedDuplicate to startAndEnd
+            }
+        }
+
+    val geocodedSplitTargets = splitTargetLocationTracks
+        .mapNotNull { (locationTrack, alignment) ->
+            getAlignmentStartAndEndM(geocodingContext, alignment)?.let { startEnd ->
+                locationTrack to startEnd
+            }
+        }
+
+    return geocodedUnusedDuplicates.map { (duplicate, duplicateStartAndEnd) ->
+        geocodedSplitTargets
+            .fold(
+                MostOverlappingLocationTrackReference()
+            ) { currentBest, (splitTarget, splitTargetStartEnd) ->
+
+                if (currentBest.overlapPercentage > 99.9) {
+                    currentBest
+                } else {
+                    val overlapStart = maxOf(duplicateStartAndEnd.start, splitTargetStartEnd.start)
+                    val overlapEnd = minOf(duplicateStartAndEnd.end, splitTargetStartEnd.end)
+
+                    val overlap = maxOf(0.0, overlapEnd - overlapStart)
+                    val intervalLength = duplicateStartAndEnd.end - duplicateStartAndEnd.start
+
+                    val overlapPercentage = overlap / intervalLength * 100
+                    if (overlapPercentage > currentBest.overlapPercentage) {
+                        MostOverlappingLocationTrackReference(
+                            locationTrack = splitTarget,
+                            overlapPercentage = overlapPercentage,
+                        )
+                    } else {
+                        currentBest
+                    }
+                }
+
+            }.let { bestNewLocationTrackReference ->
+                val newReferenceTrackId = bestNewLocationTrackReference.locationTrack?.id as IntId?
+                checkNotNull(newReferenceTrackId) {
+                    "Could not find a new location track reference for an unused duplicate!"
+                }
```

Comment:
Tulkitsin kriittiseksi virheeksi, mutta onkohan tämä oikeasti näin?

Reply:
Tähän tilanteeseen taidetaan päästä ainoastaan silloin jos/kun käyttämättömät duplikaattiraiteet ovat kokonaan splitattavan raiteen alueen ulkopuolella? Tuo tilannehan pitää kyllä käsitellä jotenkin, ja täällä heittäminen voi olla ihan fiksua. Mutta oikeastaan tästä herää kysymys, että kuuluuko tuollaista raidetta pystyä splittaamaan alkujaansakaan 🤔 Mielestäni tuo voisi hyvin olla samanlainen koko splittauksen aloittamisen estävä tilanne kuin esim. se, että duplikaatteja on väärällä ratanumerolla.